app/models/course_user.rb

```diff
@@ -23,6 +24,15 @@ class CourseUser < ApplicationRecord
   # A set of roles which comprise the managers of a course.
   MANAGER_ROLES = Set[:manager, :owner].map { |v| roles[v] }.freeze
 
+  ASSOCIATED_MODELS = %i[
```

Comment:
Are these associated models exhaustive? How about course_discussion_posts, submissions, video_submissions, activities, etc?

Are we sure we want to cascade delete / soft-delete the associated models in the first place?

Reply:
Please confirm if you have also checked the effects on jobs or not?

Reply:
I have updated to skip cascading soft-delete on associated models, so we don't need to worry about side effects from this.

Reply:
This variable `ASSOCIATED_MODELS` is unused, it should not exist.